src/OverlayLoader.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * ******************************************************************************
+ * Copyright (c) 2021 Robert Bosch GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/epl-2.0/index.php
+ *
+ * *****************************************************************************
+ */
+
+/* helper fuctions to load overlays during server startup */
+
+#include <stdexcept>
+#include <jsoncons/json.hpp>
+
+#include "OverlayLoader.hpp"
+#include "kuksa.pb.h"
+
+using jsoncons::json;
+
+std::vector<boost::filesystem::path> gatherOverlays(
+    std::shared_ptr<ILogger> logger, boost::filesystem::path overlaydir) {
+  std::vector<boost::filesystem::path> overlayfiles;
+  logger->Log(LogLevel::VERBOSE,
+              "Searching overlays in \"" + overlaydir.generic_string() + "\"");
+  if (!boost::filesystem::is_directory(overlaydir)) {
+    throw std::runtime_error("Overlays need to be a directory. \"" +
+                             overlaydir.generic_string() + "\" is not.");
+  }
+  for (auto &p : boost::filesystem::directory_iterator(overlaydir)) {
+    if (boost::filesystem::is_regular_file(p.path()) &&
+        p.path().extension() == ".json") {
+      logger->Log(LogLevel::VERBOSE, "Found " + p.path().generic_string());
+      overlayfiles.push_back(p);
+    }
+  }
+
+  // Sort according to filename
+  std::sort(overlayfiles.begin(), overlayfiles.end(),
```

Comment:
why sort?

Reply:
Because the resulting VSS depends on the order the VSS files are merged. So the documentation states will be merged in "alphanumeric order". This sorts the detected filenames (it is not defined in what order we get them, when just iterating directory entries)